### PR TITLE
Introduce Nebari support in code generator

### PR DESCRIPTION
**Pull request by guptaaryan16**
### Description
The PR aims to provide [Nebari](https://www.nebari.dev) support in the code generator app. Nebari is a great infrastructure management tool for data scientists and companies and can greatly help in management of cloud infrastructure by providing a JupyterHub like environment and instances.
This PR uses [Jupyterlab-Github](https://github.com/jupyterlab/jupyterlab-github) extension for connecting and fetching the repositories committed by the netlify functions by the hashes generated.

### Additional context

This issue was discussed in our weekly meeting this week.

---

### What is the purpose of this pull request?

- [ ] Bug fix
- [x] New feature
- [ ] Other

### CODE REVIEW

**Comment by vfdev-5** [comment]
@guptaaryan16 thanks for the PR, few things about UI just came : 
- let's add a link to Nebari site : https://www.nebari.dev/ 
  - I would add a small section before "Requirements":
- Let's add splitting horizontal lines between sections
- Let's make the button larger and make it open a new tab with the link

```
[Nebari](https://www.nebari.dev/): open source data science platform
---
Requirements:
...  
---
Nebari HubURL:
...
Nebari Username
...

[Large Button to Open New Tab]
```

File: functions/utils.js [hunk 1]
```diff
@@ -85,10 +90,14 @@ export async function getZip_Uid(data) {
   }
 }
 
-export function getRootUrlWithoutTrailingSlash(url) {
+/**
+ * Create a file on GitHub with Octokit.
+ * @param {string} url
+ * @returns {string} rootUrl
+ */
+export async function getRootUrlWithoutTrailingSlash(url) {
```
Comment (vfdev-5) [inline functions/utils.js:98]: why this should be an async function ?

File: src/components/NavNebari.vue [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+              ></path>
+            </svg>
+          </span>
+        </div>
+      </button>
+      <div class="dropdown-content">
+        <div class="text-box-nebari" id="nebari-text">
+          <span class="sub-text">
+            <a href="https://www.nebari.dev/"><strong>Nebari</strong></a
+            >: open source data science platform
```
Comment (vfdev-5) [inline src/components/NavNebari.vue:41]: Let's make all the text as a link
```suggestion
            <a href="https://www.nebari.dev/">
              <strong>Nebari</strong>: open source data science platform
            </a>  
```

File: src/components/NavNebari.vue [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+            >
+            extension on your Nebari instance.
+          </p>
+          <pre class="code-snippet">pip install jupyterlab-github</pre>
+        </div>
+
+        <hr class="solid" />
+        <div class="text-input-group">
+          <div class="text-box-nebari">
+            <h5 class="form-label">Nebari HubURL</h5>
```
Comment (vfdev-5) [inline src/components/NavNebari.vue:60]:
```suggestion
            <h5 class="form-label">Nebari URL</h5>
```

File: functions/utils.js (hunk 1 above)
Comment (guptaaryan16) [reply functions/utils.js:98]: @vfdev-5 I tested the netlify function locally and it was giving some error so I wanted to test this here in production, if it doesn't work lets revert it

File: src/components/NavNebari.vue [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+            <input
+              v-model="hubUrl"
+              @input="validateHubUrl"
+              placeholder="https://myjupyterhub.org"
```
Comment (vfdev-5) [inline src/components/NavNebari.vue:65]:
```suggestion
              placeholder="https://nebari.yourdomain.dev"
```

**Comment by guptaaryan16** [comment]
I have tested the app and it seems to work now as expected.

File: functions/nebari.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+  ]
+
+  const argparser = data.argparser
+  const execution_nb_commands = [
+    `!python main.py ${
+      argparser === 'hydra'
+        ? '#--config-dir=/content/ --config-name=config.yaml'
+        : 'config.yaml'
+    }`
+  ]
```
Comment (vfdev-5) [inline functions/nebari.js:66]: Can we refactor these methods with those from colab.js to make less code duplications?

**Review by vfdev-5 — APPROVED** [review]
Thanks @guptaaryan16, looks good to me!
Please check if colab and nebari buttons are still working

**Pull request merged**